piece_of_cake.py

```diff
@@ -0,0 +1,17 @@
+def get_recipe_price(prices=None, optionals=None, **ingredients):
+
+    if prices is not None and ingredients is not None and len(ingredients) == len(prices):
+        if optionals is None:
+            optionals = []
+
+        shopping_cart = 0
+        for price in prices:
+            if price not in optionals:
+                num_for_mul = ingredients[price]/100
+                shopping_cart = shopping_cart + int(prices[price])*num_for_mul
+        print(int(shopping_cart))
+    else:
```

Comment:
You dont have to validate input - if you request parameters and the user doesnt supplies them, or supplies other types, its their problem, and the code will fail  - you have to handle all the legit values - an empty dictionary of ingredients is a legit value - None is not  a legit value

Reply:
Another legit option - what happens if an ingredient is not in prices